### PR TITLE
Add kernel testing steps, update release docs

### PR DESCRIPTION
**Pull request by legoktm**
## Status

Ready for review

## Description of Changes

* Description:
  * Add Linux kernel testing steps
  * Remove kernel mentions from the release process

* Related Issues
  * https://github.com/freedomofpress/securedrop/issues/6514
  * https://github.com/freedomofpress/securedrop/issues/6328

## Testing
* Visual review

## Release 

* No special considerations

## Checklist (Optional)

- [x] Doc linting (`make docs-lint`) passed locally
- [ ] Doc link linting (`make docs-linkcheck`) passed
- [x] You have previewed (`make docs`) docs at http://localhost:8000

### CODE REVIEW

**Comment by legoktm** [comment]
(rebased after the files moved)

File: docs/kernel.rst [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+--------------------
+
+The following steps should be performed for all of the `recommended hardware`_:
+
+#. Install the new kernel packages on your *Monitor Server*, reboot. Verify with ``uname -r`` that you are using the new kernel.
+#. If it doesn't boot, see the `Troubleshooting Kernel Updates`_ documentation.
+#. Verify ``paxtest`` returns expected results.
+#. Verify the `spectre-meltdown-checker`_ returns expected results.
+#. Upgrade your *Application Server* to the new kernel and reboot.
+#. Run basic smoke tests of SecureDrop, verify you can send a submission and a journalist can reply.
```
Comment (nathandyer) [inline docs/kernel.rst:19]: We should clarify how to run basic smoke tests here, or at least link out to more details.

Also, I might suggest a slight adjustment to the wording just to make it flow a bit better:

"Run basic smoke tests of SecureDrop, then verify you can send a submission and that a journalist can reply."

File: docs/kernel.rst [hunk 2]
```diff
@@ -0,0 +1,25 @@
+Linux kernel maintenance
+========================
+
+We build and publish our own Linux kernels with additional
+`grsecurity hardening patches`_.
+The `kernel-builder`_ repository contains scripts that fetches upstream
```
[the remark this answers is not in the document]
Comment (legoktm) [reply docs/kernel.rst:6]: Done.

File: docs/kernel.rst [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+kernel tarballs plus grsecurity patches and produces Debian packages.
+
+Testing a new kernel
+--------------------
+
+The following steps should be performed for all of the `recommended hardware`_:
+
+#. Install the new kernel packages on your *Monitor Server*, reboot. Verify with ``uname -r`` that you are using the new kernel.
```
[the remark this answers is not in the document]
Comment (legoktm) [reply docs/kernel.rst:14]: Done

File: docs/kernel.rst [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+#. If it doesn't boot, see the `Troubleshooting Kernel Updates`_ documentation.
+#. Verify ``paxtest`` returns expected results.
+#. Verify the `spectre-meltdown-checker`_ returns expected results.
```
[the remark this answers is not in the document]
Comment (legoktm) [reply docs/kernel.rst:17]: That's much better, done.

File: docs/kernel.rst (hunk 1 above)
Comment (legoktm) [reply docs/kernel.rst:19]: Those are supposed to be the smoke tests. Switched to "Run basic smoke tests of SecureDrop by verifying you can send a submission..."

Comment (nathandyer) [reply docs/kernel.rst]: Ah, perfect! Much clearer now, thanks. :)

**Review by nathandyer — APPROVED** [review]

**Comment by nathandyer** [comment]
LGTM!

**Comment by maeve-fpf** [comment]
Hey @legoktm @nathandyer, looks like this build is failing and so didn't deploy: https://g.codefresh.io/build/633d9603c6029c0311ca9de1 (you may not be able to access, sorry) - I was able to reproduce locally:

```
Transcript written on SecureDropDevDocs.log.
Latexmk: Index file 'SecureDropDevDocs.idx' was written
Latexmk: Missing input file: '{/docs/_build/doctrees/images/e8339822932e8b554afc4b4967a9e3f3de82ab43/horizontal-auto}.svg' from line
  'LaTeX Warning: File `{/docs/_build/doctrees/images/e8339822932e8b554afc4b4967a9e3f3de82ab43/horizontal-auto}.svg' not found on input line 387.'
Latexmk: Missing input file: '{/docs/_build/doctrees/images/e8339822932e8b554afc4b4967a9e3f3de82ab43/horizontal-auto}.svg' from line
  'LaTeX Warning: File `{/docs/_build/doctrees/images/e8339822932e8b554afc4b4967a9e3f3de82ab43/horizontal-auto}.svg' not found on input line 387.'
Failure to make 'SecureDropDevDocs.pdf'
Latexmk: Examining 'SecureDropDevDocs.log'
=== TeX engine is 'pdfTeX'
Latexmk: Errors, in force_mode: so I tried finishing targets
Collected error summary (may duplicate other messages):
  pdflatex: Command for 'pdflatex' gave return code 1
      Refer to 'SecureDropDevDocs.log' for details
make[1]: *** [Makefile:29: SecureDropDevDocs.pdf] Error 12
make[1]: Leaving directory '/docs/_build/latex'
make: *** [Makefile:25: latexpdf] Error 2
Removing intermediate container 2ba4212dcaa8
 ---> a6268406dc35
Step 6/11 : FROM nginx:mainline-alpine@sha256:2959a35e1b1e61e2419c01e0e457f75497e02d039360a658b66ff2d4caab19c4
 ---> 804f9cebfdc5
Step 7/11 : COPY deploy/nginx.conf /etc/nginx
 ---> Using cache
 ---> 8b09ca322805
Step 8/11 : RUN mkdir -p /opt/nginx/run /opt/nginx/webroot/en/latest && chown -R nginx:nginx /opt/nginx
 ---> Using cache
 ---> 5156f9a395af
Step 9/11 : USER nginx
 ---> Using cache
 ---> 800a23ec253b
Step 10/11 : COPY --from=sphinx --chown=nginx:nginx build/latest/html/ /opt/nginx/webroot/en/latest/
 ---> 0223533e5abc
Step 11/11 : COPY --from=sphinx --chown=nginx:nginx build/latest/latex/SecureDropDevDocs.pdf /opt/nginx/webroot/en/latest/
COPY failed: stat build/latest/latex/SecureDropDevDocs.pdf: file does not exist
```

Haven't done more investigation yet.

**Comment by maeve-fpf** [comment]
Looks like older commits are failing for me with the same error, even though yesterday the last PR was successful: https://freedomofpress.slack.com/archives/CABPD6ANB/p1664898223951189

**Comment by legoktm** [comment]
Fixed in https://github.com/freedomofpress/securedrop-dev-docs/pull/23.

**Pull request merged**